algorithms/FaceApi/FaceIdentifier.cs

```diff
@@ -0,0 +1,122 @@
+using System;
+using System.IO;
+using System.Linq;
+using System.Collections.Generic;
+using System.Threading.Tasks;
+using Microsoft.Azure.CognitiveServices.Vision.Face;
+using Microsoft.Azure.CognitiveServices.Vision.Face.Models;
+using RateLimiter;
+
+namespace FaceApi
+{
+    public class FaceIdentifier
+    {
+        private TimeLimiter RateLimit { get; }
+        private FaceClient Client { get; }
+
+        public FaceIdentifier(string apiKey, string apiEndpoint)
+        {
+            Client = new FaceClient(new ApiKeyServiceClientCredentials(apiKey))
+            {
+                Endpoint = apiEndpoint
+            };
+
+            RateLimit = TimeLimiter.GetFromMaxCountByInterval(10, TimeSpan.FromSeconds(1));
+        }
+
+        public async Task<bool> Evaluate(string groupId, double matchThreshold, string imagePath1, string imagePath2)
+        {
+            var faces = await Task.WhenAll(new[] { DetectFaces(imagePath1), DetectFaces(imagePath2) });
+            var result = await Client.Face.VerifyFaceToFaceAsync(faces[0][0], faces[1][0]);
```

Comment:
Would it be possible to change this so that it checks all faces in both images and if any pair is a match returns true? That way for testing we can use images that have multiple people in them.

Reply:
I thought this would be handled during the alignment step prior to running the evaluation? I just pushed an evaluation function that computes accuracy/precision/recall given a pairs.txt file. Check it out: 7d5d577.

Reply:
This might still be useful though for the custom data-set (where we don't necessarily know exactly who are all the people in the image), so I've submitted an issue for this https://github.com/c-w/faceanalysis/issues/38.